From 7f9c6109f78762675698d1508dd5895ad0a4e85f Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:46 -0400
Subject: [iommu] Change iommu driver to call map trace event

Message-id: <20140810182046.24055.86026.stgit@gir.stowe>
Patchwork-id: 87584
O-Subject: [RHEL7.1 PATCH 13/19] iommu: Change iommu driver to call map trace event
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: e0be7c867882320b65714a4386760382196dc7e8

commit e0be7c867882320b65714a4386760382196dc7e8
Author: Shuah Khan <shuah.kh@samsung.com>
Date:   Thu Aug 15 11:59:28 2013 -0600

    iommu: Change iommu driver to call map trace event

    Change iommu driver to call map trace event. This iommu_map_unmap class even
    can be enabled to trigger when iommu map iommu ops is called. Trace informat
    includes iova, physical address (map event only), and size.

    Testing:
    Added trace calls to iommu_prepare_identity_map() for testing some of the
    conditions that are hard to trigger. Here is the trace from the testing:

           swapper/0-1     [003] ....     1.854102: map: IOMMU: iova=0x00000000c

    Signed-off-by: Shuah Khan <shuah.kh@samsung.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index b630754..ea49fe8 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -842,6 +842,8 @@ int iommu_map(struct iommu_domain *domain, unsigned long iova,
  /* unroll mapping in case something went wrong */
  if (ret)
   iommu_unmap(domain, orig_iova, orig_size - size);
+ else
+  trace_map(iova, paddr, size);
 
  return ret;
 }
-- 
1.7.1